src/main.rs

```diff
@@ -486,6 +491,43 @@ fn run() -> Result<()> {
                 )
             })?;
 
+            if app_matches.is_present("list-languages") {
+                let languages = assets.syntax_set.syntaxes();
+
+                let longest = languages.iter()
+                    .map(|s| s.name.len())
+                    .max()
+                    .unwrap_or(32); // Fallback width if they have no language definitions.
+
+                let separator = " | ";
+                for lang in languages {
+                    print!("{:width$}{}", lang.name, separator, width = longest);
+
+                    // Line-wrapping for the possible file extension overflow.
+                    let desired_width = 48;
```

Comment:
Maybe this could be `desired_width = term_width - longest - padding`?

Reply:
How should I get the term_width? I looked around and I'm not seeing a function I can call to get that value.

I notice it's set with a literal `90` in `run()`; should I just create a constant and apply it to both places?

Reply:
The `term_width` is computed/retrieved in `printer.s`. We should probably move that to the `run` function and store it in the `Options` struct.